include/bout/region.hxx

```diff
@@ -62,55 +100,160 @@
+    ind += n.ind;
+    return *this;
+  }
+
+  /// In-place subtraction
+  SpecificInd &operator-=(SpecificInd n) {
+    ind -= n.ind;
+    return *this;
+  }
+
+  /// Modulus operator
+  SpecificInd operator%(int n) {
+    SpecificInd new_ind(ind % n);
+    return new_ind;
+  }
 };
-// Make identical subclasses with different names
+
+/// Relational operators
+inline bool operator==(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return lhs.ind == rhs.ind;
+}
+inline bool operator!=(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return !operator==(lhs, rhs);
+}
+inline bool operator<(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return lhs.ind < rhs.ind;
+}
+inline bool operator>(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return operator<(rhs, lhs);
+}
+inline bool operator>=(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return !operator<(lhs, rhs);
+}
+inline bool operator<=(const SpecificInd &lhs, const SpecificInd &rhs) {
+  return !operator>(lhs, rhs);
+}
+
+/// Arithmetic operators with integers
+inline SpecificInd operator+(SpecificInd lhs, const SpecificInd &rhs) { return lhs += rhs; }
+inline SpecificInd operator+(SpecificInd lhs, int n) { return lhs += n; }
+inline SpecificInd operator+(int n, SpecificInd rhs) { return rhs += n; }
+inline SpecificInd operator-(SpecificInd lhs, int n) { return lhs -= n; }
+inline SpecificInd operator-(SpecificInd lhs, const SpecificInd &rhs) { return lhs -= rhs; }
+
+/// Index-type for `Field3D`s
 class Ind3D : public SpecificInd {
 public:
   Ind3D() : SpecificInd(){};
   Ind3D(int i) : SpecificInd(i){};
+
   // Note operator= from base class is always hidden
   // by implicit method so have to be explicit
   Ind3D &operator=(int i) {
     ind = i;
     return *this;
   }
+
+  Ind3D &operator=(SpecificInd i) {
+    ind = i.ind;
+    return *this;
+  }
 };
+
+/// Index-type for `Field2D`s
 class Ind2D : public SpecificInd {
 public:
   Ind2D() : SpecificInd(){};
   Ind2D(int i) : SpecificInd(i){};
+
   Ind2D &operator=(int i) {
     ind = i;
     return *this;
   }
+
+  Ind2D &operator=(SpecificInd i) {
+    ind = i.ind;
+    return *this;
+  }
 };
 
+/// Specifies a set of indices which can be iterated over and begin()
+/// and end() methods for range-based for loops.
+///
+/// `Region` is templated on either `Ind2D` or `Ind3D` for `Field2D`s
+/// or `Field3D`s, respectively. Trying to create a `Region` using any
+/// other type is an error.
+///
+/// The set of indices is also broken down into sets of contiguous
+/// blocks of at most MAXREGIONBLOCKSIZE indices. This allows loops to
+/// be parallelised with OpenMP. Iterating using a "block region" may
+/// be more efficient, although it requires a bit more set up. The
+/// helper macro BLOCK_REGION_LOOP is provided to simplify things.
+///
+/// Example
+/// -------
+///
+/// Index ranges can be defined manually:
+///
+///     Region<Ind3D>::RegionIndices region {0, 2, 4, 8};
```

Comment:
Does this create a region with only these four indices, or one with two ranges: 0,1,2  and 4 5 6 7 8 ? (or maybe 0,1 and 4,5,6,7?)

Reply:
Ah ok. Maybe just a note in the comments then; the "Index ranges" part of the above comment had me a bit confused. Thanks!

Reply:
I've exapanded this comment now, so it's hopefully a bit clearer.